core/realm.rs

```diff
@@ -43,7 +43,7 @@ pub(crate) struct ContextState {
   pub(crate) js_format_exception_cb: Option<Rc<v8::Global<v8::Function>>>,
   pub(crate) js_wasm_streaming_cb: Option<Rc<v8::Global<v8::Function>>>,
   pub(crate) pending_promise_rejections:
-    HashMap<v8::Global<v8::Promise>, v8::Global<v8::Value>>,
+    LinkedHashMap<v8::Global<v8::Promise>, v8::Global<v8::Value>>,
```

Comment:
Does `IndexMap` not work for this?

Reply:
[Its documentation](https://docs.rs/indexmap/1.9.2/indexmap/map/struct.IndexMap.html#method.shift_remove) states that removing an entry while preserving insertion order takes O(n) time on average.

Reply:
`indexmap` is already a dep of core, let's use it.

Reply:
I think the current solution is fine, it allows to get rid of the nasty:
```rust
    let key = {
      context_state
        .pending_promise_rejections
        .keys()
        .next()
        .unwrap()
        .clone()
    };
    let handle = context_state
      .pending_promise_rejections
      .remove(&key)
      .unwrap();
```